desc/objectives/_wrappers.py

```diff
@@ -229,27 +230,19 @@ def jac(self, x):
         # Jacobian matrices wrt combined state vectors
         Fx = self._eq_objective.jac(xf)
         Gx = self._objective.jac(xg)
-        Fx = {
-            arg: Fx[:, self._eq_objective.x_idx[arg]] for arg in self._eq_objective.args
-        }
-        Gx = {arg: Gx[:, self._objective.x_idx[arg]] for arg in self._objective.args}
-        for arg in self._eq_objective.args:
-            if arg not in Fx.keys():
-                Fx[arg] = jnp.zeros((self._eq_objective.dim_f, self.dimensions[arg]))
-            if arg not in Gx.keys():
-                Gx[arg] = jnp.zeros((self._objective.dim_f, self.dimensions[arg]))
-        Fx = jnp.hstack([Fx[arg] for arg in arg_order if arg in Fx])
-        Gx = jnp.hstack([Gx[arg] for arg in arg_order if arg in Gx])
+        Fx = align_jacobian(Fx, self._eq_objective, self._objective)
```

Comment:
this is just a refactor of the old code?

Reply:
yeah we were doing this in a few places and will probably need to also do it for more multiobjective/constrained stuff so made sense to just roll it into a utility function.